ui/lib/kv/addon/components/page/secret/metadata-details.hbs

```diff
@@ -7,6 +7,59 @@
   </:tabLinks>
 
   <:toolbarActions>
-    <ToolbarLink @route="secret.metadata.edit">Edit metadata</ToolbarLink>
+    {{! You must have update on metadata, create is not enough to edit. }}
+    {{#if @model.canUpdateMetadata}}
+      <ToolbarLink @route="secret.metadata.edit" data-test-edit-metadata>Edit metadata</ToolbarLink>
+    {{/if}}
   </:toolbarActions>
-</KvPageHeader>
\ No newline at end of file
+</KvPageHeader>
+
+<div class="form-section">
+  <label class="title is-5 has-top-padding-m">
+    Custom metadata
+  </label>
+</div>
+<div class="box is-fullwidth is-sideless is-paddingless is-marginless">
+  {{! wrap the whole conditional in each-in because there is no easy way to check if an object is empty in handlebars }}
+  {{#each-in (or @model.customMetadata @secretCustomMetadata) as |key value|}}
+    <InfoTableRow @alwaysRender={{false}} @label={{key}} @value={{value}} data-test-custom-metadata />
+  {{else if (or @model.canReadMetadata)}}
```

Comment:
I think this empty state should be moved to the `{{else}}` block because this means the metadata is empty, right?

Reply:
So these conditionals suck. But if I make these changes this is what happens because the Empty state is out of the #each-in. When I work through this in my head, I have to keep all the empty states in the each-in for this to work:
* Here's my 🧠 pattern: Do I return secret metadata? If I don't, is it because I don't have access or is it because none exists? 
![image](https://github.com/hashicorp/vault/assets/6618863/0f55e73b-7440-4d70-bae6-9218e894f72c)

Reply:
Okay, took me a moment to work that one through my head. But yep, that reads clearer.